enhancements/cluster-logging/azure_workload_identity_auth.md

```diff
@@ -0,0 +1,264 @@
+---
+title: workload_identity_support_for_azure_monitor_logs
```

Comment:
nit:  hyphens, rather than snakecase, is the standard for filenames in this directory.   
suggested title change:   "Workload Identity Auth for Azure Monitor Logs" as I don't think the "Entra ID" part is necessary to mention in the PR or anywhere else.   Entra is their entire suite of IAM and access solutions.  It's implied.